ncache-vm-cluster/azuredeploy.json

```diff
@@ -0,0 +1,401 @@
+      "type": "string",
+      "metadata": {
+        "description": "Last Name"
+      }
+    },
+    "company": {
+      "type": "string",
+      "metadata": {
+        "description": "Company"
+      }
+    },
+    "emailAddress": {
+      "type": "string",
+      "metadata": {
+        "description": "Email Address"
+      }
+    },
+    "clusterName": {
+      "type": "string",
+      "metadata": {
+        "description": "Name for cache cluster"
+      }
+    },
+    "clusterPort": {
+      "type": "int",
+      "minValue": 1024,
+      "maxValue": 65535,
+      "defaultValue": 7800,
+      "metadata": {
+        "description": "Port for NCache Cluster"
+      }
+    },
+    "cacheTopology": {
+      "type": "string",
+      "defaultValue": "PartitionedOfReplica",
+      "metadata": {
+        "description": "Topology for NCache Cluster"
+      }
+    },
+    "replicationStrategy": {
+      "type": "string",
+      "allowedValues": [
+        "async",
+        "sync"
+      ],
+      "defaultValue": "async"
+    },
+    "cacheSize": {
+      "type": "string",
+      "defaultValue": "1024"
+    },
+    "evictionPolicy": {
+      "type": "string",
+      "allowedValues": [
+        "none",
+        "LRU",
+        "LFU",
+        "Priority"
+      ],
+      "defaultValue": "LRU"
+    },
+    "evictionPercentage": {
+      "type": "string",
+      "defaultValue": "5"
+    },
+    "publicIpAddressName": {
+      "type": "string",
+      "defaultValue": "[concat(parameters('virtualMachineNamePrefix'), 'ip')]"
+    },
+    "location": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().location]"
+    },
+    "virtualMachineNamePrefix": {
+      "type": "string",
+      "defaultValue": "NCacheVM"
+    },
+    "virtualMachineSize": {
+      "type": "string",
+      "defaultValue": "Standard_DS2_v2"
+    },
+    "adminUsername": {
+      "type": "string",
+      "defaultValue": "azadmin"
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "defaultValue": "123Islamabad"
+    },
+    "addressPrefix": {
+      "type": "string",
+      "defaultValue": "10.0.0.0/16"
+    },
+    "subnetName": {
+      "type": "string",
+      "defaultValue": "Subnet1"
+    },
+    "subnetPrefix": {
+      "type": "string",
+      "defaultValue": "10.0.0.0/24"
+    },
+    "virtualNetworkName": {
+      "type": "string",
+      "defaultValue": "MyVNET1"
+    },
+    "networkInterfaceName": {
+      "type": "string",
+      "defaultValue": "[concat(parameters('virtualMachineNamePrefix'), 'NIC')]"
+    },
+    "networkSecurityGroupName": {
+      "type": "string",
+      "defaultValue": "NetworkSecurityGroup"
+    },
+    "NCacheClusterCreationScriptFileUri": {
+      "type": "string",
+      "defaultValue": "NotSpecified"
+    }
+  },
+  "resources": [
+    {
+      "name": "[concat(parameters('virtualMachineNamePrefix'), copyIndex(1))]",
+      "type": "Microsoft.Compute/virtualMachines",
+      "apiVersion": "2017-12-01",
```

Comment:
update apiVersions throughout - nothing more than 2 years old (unless it's the latest)